src/Sylius/Behat/Context/Api/Admin/ManagingProductAssociationTypesContext.php

```diff
@@ -0,0 +1,354 @@
+    }
+
+    /**
+     * @Then I should see the product association type :name in the list
+     * @Then this product association type should still be named :name
+     */
+    public function iShouldSeeTheProductAssociationTypeInTheList(string $name): void
+    {
+        Assert::true(
+            $this->responseChecker->hasItemWithValue($this->client->index(), 'name', $name),
+            sprintf('There is no product association type with name "%s"', $name)
+        );
+    }
+
+    /**
+     * @When I delete the :productAssociationType product association type
+     */
+    public function iDeleteTheProductAssociationType(ProductAssociationTypeInterface $productAssociationType): void
+    {
+        $this->client->delete($productAssociationType->getCode());
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully deleted
+     * @Then I should be notified that they have been successfully deleted
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyDeleted(): void
+    {
+        Assert::true($this->responseChecker->isDeletionSuccessful(
+            $this->client->getLastResponse()),
+            'Product association type could not be deleted'
+        );
+    }
+
+    /**
+     * @Then /^(this product association type) should no longer exist in the registry$/
+     */
+    public function thisProductAssociationTypeShouldNoLongerExistInTheRegistry(ProductAssociationTypeInterface $productAssociationType): void
+    {
+        Assert::false(
+            $this->responseChecker->hasItemWithValue($this->client->index(), 'code', $productAssociationType->getCode()),
+            sprintf('Product association type with code %s exist', $productAssociationType->getCode())
+        );
+    }
+
+    /**
+     * @When I want to modify the :productAssociationType product association type
+     */
+    public function iWantToModifyTheProductAssociationType(ProductAssociationTypeInterface $productAssociationType): void
+    {
+        $this->client->buildUpdateRequest($productAssociationType->getCode());
+    }
+
+    /**
+     * @When I rename it to :name in :language
+     */
+    public function iRenameItToIn(string $name, string $language): void
+    {
+        $this->client->updateRequestData(['translations' => [$language => ['name' => $name, 'locale' => $language]]]);
+    }
+
+    /**
+     * @When I save my changes
+     * @When I try to save my changes
+     */
+    public function iSaveMyChanges(): void
+    {
+        $this->client->update();
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully edited
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyEdited(): void
+    {
+        Assert::true(
+            $this->responseChecker->isUpdateSuccessful($this->client->getLastResponse()),
+            'Product association type could not be edited'
+        );
+    }
+
+    /**
+     * @Then this product association type name should be :name
```

Comment:
Same as https://github.com/Sylius/Sylius/pull/11319/files#diff-9f9847e886745cd71df41b26b4816ae1R166
```suggestion
     * @Then /^(this product association type) name should be "([^"]+)"$/
```